File: leetcode/rust/reverse-string.rs
impl Solution1 {
    pub fn reverse_string(s: &mut Vec<char>) {
        s.reverse();
    }
}

impl Solution2 {
    pub fn reverse_string(s: &mut Vec<char>) {
        let mut left = 0;
        let mut right = s.len() - 1;

        while (left < right) {
            s.swap(left, right);
            left += 1;
            right -= 1;
        }
    }
}

impl Solution3 {
    pub fn reverse_string(s: &mut Vec<char>) {
        let mut left = 0;
        let mut right = s.len() - 1;

        while (left < right) {
            (s[left], s[right]) = (s[right], s[left]);
            left += 1;
            right -= 1;
        }
    }
}

File: leetcode/python/reverse-string.py
class Solution1:
    def reverseString(self, s: list[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """

        s.reverse()

class Solution2:
    def reverseString(self, s: list[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        l, r = 0, len(s)-1

        while l < r:
            s[l], s[r] = s[r], s[l]
            l += 1
            r -= 1